Modules/_testcapi/tuple.c

```diff
@@ -2,14 +2,107 @@
 #include "util.h"
 
 
+static PyObject *
+tuple_get_size(PyObject *Py_UNUSED(module), PyObject *obj)
+{
+    NULLABLE(obj);
+    RETURN_SIZE(PyTuple_GET_SIZE(obj));
+}
+
+static PyObject *
+tuple_get_item(PyObject *Py_UNUSED(module), PyObject *args)
+{
+    PyObject *obj;
+    Py_ssize_t i;
+    if (!PyArg_ParseTuple(args, "On", &obj, &i)) {
+        return NULL;
+    }
+    NULLABLE(obj);
+    return Py_XNewRef(PyTuple_GET_ITEM(obj, i));
+}
+
+static PyObject *
+tuple_set_item(PyObject *Py_UNUSED(module), PyObject *args)
+{
+    PyObject *obj, *value, *newtuple;
+    Py_ssize_t i;
+    if (!PyArg_ParseTuple(args, "OnO", &obj, &i, &value)) {
+        return NULL;
+    }
+    NULLABLE(value);
+    if (PyTuple_CheckExact(obj)) {
+        Py_ssize_t size = PyTuple_Size(obj);
+        newtuple = PyTuple_New(size);
+        if (!newtuple) {
+            return NULL;
+        }
+        for (Py_ssize_t n = 0; n < size; n++) {
+            PyTuple_SET_ITEM(newtuple, n, Py_XNewRef(PyTuple_GET_ITEM(obj, n)));
+        }
+    }
+    else {
+        NULLABLE(obj);
+        newtuple = obj;
+    }
+    PyObject *val = PyTuple_GET_ITEM(newtuple, i);
+    PyTuple_SET_ITEM(newtuple, i, Py_XNewRef(value));
+    Py_DECREF(val);
+    return newtuple;
+}
+
+static PyObject *
+_tuple_resize(PyObject *Py_UNUSED(module), PyObject *args)
+{
+    PyObject *tup;
+    Py_ssize_t newsize;
+    int new = 1;
+    if (!PyArg_ParseTuple(args, "On|p", &tup, &newsize, &new)) {
+        return NULL;
+    }
+    if (new) {
+        Py_ssize_t size = PyTuple_Size(tup);
+        PyObject *newtup = PyTuple_New(size);
+        for (Py_ssize_t n = 0; n < size; n++) {
+            PyTuple_SET_ITEM(newtup, n, Py_XNewRef(PyTuple_GET_ITEM(tup, n)));
+        }
+        tup = newtup;
+    }
+    else {
+        Py_XINCREF(tup);
+    }
+    int r = _PyTuple_Resize(&tup, newsize);
+    if (r == -1) {
+        assert(tup == NULL);
+        return NULL;
+    }
+    return tup;
+}
+
+static PyObject *
+_check_item_is_NULL(PyObject *Py_UNUSED(module), PyObject *args)
```

Comment:
since the first argument is a tuple, I suggest to rename the function to: _tuple_check_item_is_NULL().

Reply:
Done.